### ISSUE TITLE
B200 and B300 incompatibility problem within axolotl framework

### ISSUE THREAD

**Issue by RenIII2211** [axolotl-ai-cloud/axolotl#3336]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Well, i don't know how it should be.
But i guess it should be running without error that stopping it mid run.

Btw if we didn't use liger and torch compile set to false it run okey, so it seems to be problem with trition but i could be wrong as someone who didn't have that much knowledge in this field.

From my limited knowledge and interaction with LLM it seems that it have to do with Triton, PTX code, PTXAS

### Current behaviour

[Axolotl_Compatibility_Problem.txt](https://github.com/user-attachments/files/24365676/Axolotl_Compatibility_Problem.txt)

### Steps to reproduce

My own experience

1. Get B300
2. Get docker run on it (axolotlai/axolotl-cloud:main-20251228-py3.11-cu128-2.8.0)
3. Setup as normally how it is if we want to do train
4. (There two set i use to test)
 
- Provide setup with all liger options and enable it.
Example:
base_model: Goekdeniz-Guelmez/Josiefied-Qwen3-4B-Instruct-2507-gabliterated-v2
trust_remote_code: true
tokenizer_use_fast: true
tokenizer_type: AutoTokenizer
model_type: AutoModelForCausalLM
output_dir: /workspace/data/outputs/Qwen3-4B
seed: 42
pretraining_dataset:
  - path: json
    data_files:
      - /workspace/data/datasets/josiefied/*.jsonl
    text_column: text
    split: train
    type: pretrain
dataset_num_proc: 16       
dataset_keep_in_memory: true
shuffle_merged_datasets: true
dataloader_num_workers: 4 
dataloader_pin_memory: true
dataloader_prefetch_factor: 2
dataloader_drop_last: true
sequence_len: 32768
sample_packing: false
pad_to_sequence_len: true

micro_batch_size: 3      
gradient_checkpointing: true  
activation_offloading: false  
fp8: false               
bf16: true

gradient_accumulation_steps: 1
max_steps: 100
tf32: true
flash_attention: true
torch_compile: false       
optimizer: adamw_torch_fused
learning_rate: 5.0e-5
weight_decay: 0.01
max_grad_norm: 1.0
lr_scheduler: one_cycle
lr_scheduler_kwargs:
  pct_start: 0.1
  anneal_strategy: cos
  div_factor: 25.0
  final_div_factor: 10.0
  three_phase: false
warmup_ratio: 0.04
lr_groups:
  - name: embeddings
    modules:
      - model.embed_tokens.weight
    lr: 3.0e-5
    weight_decay: 0.0036
  - name: task_specific_layers
    modules:
      - model.layers.27
      - model.layers.28
      - model.layers.29
      - model.layers.30
      - model.layers.31
      - model.layers.32
      - model.layers.33
      - model.layers.34
      - model.layers.35
      - model.norm.weight
    lr: 3.5e-5
    weight_decay: 0.0060
plugins:
  - axolotl.integrations.liger.LigerPlugin
liger_rope: true
liger_rms_norm: true
liger_glu_activation: true
liger_layer_norm: true
liger_fused_linear_cross_entropy: true
liger_dropout: true
logging_steps: 1
save_steps: 100
save_total_limit: 1
include_tkps: true
use_wandb: false
special_tokens:
  pad_token: "<|endoftext|>"  

  - Provide setup with torch compile enable.
Example:
base_model: Goekdeniz-Guelmez/Josiefied-Qwen3-4B-Instruct-2507-gabliterated-v2
trust_remote_code: true
tokenizer_use_fast: true
tokenizer_type: AutoTokenizer
model_type: AutoModelForCausalLM
output_dir: /workspace/data/outputs/Qwen3-4B
seed: 42
pretraining_dataset:
  - path: json
    data_files:
      - /workspace/data/datasets/josiefied/*.jsonl
    text_column: text
    split: train
    type: pretrain
dataset_num_proc: 16 
dataset_keep_in_memory: true
shuffle_merged_datasets: true
dataloader_num_workers: 4 
dataloader_pin_memory: true
dataloader_prefetch_factor: 2
dataloader_drop_last: true
sequence_len: 32768
sample_packing: false
pad_to_sequence_len: true

micro_batch_size: 1       
gradient_checkpointing: true  
activation_offloading: false  
fp8: false                 
bf16: true

gradient_accumulation_steps: 1
max_steps: 100
tf32: true
flash_attention: true
torch_compile: true
torch_compile_backend: inductor
torch_compile_mode: default
optimizer: adamw_torch_fused
learning_rate: 5.0e-5
weight_decay: 0.01
max_grad_norm: 1.0
lr_scheduler: one_cycle
lr_scheduler_kwargs:
  pct_start: 0.1
  anneal_strategy: cos
  div_factor: 25.0
  final_div_factor: 10.0
  three_phase: false
warmup_ratio: 0.04
lr_groups:
  - name: embeddings
    modules:
      - model.embed_tokens.weight
    lr: 3.0e-5
    weight_decay: 0.0036
  - name: task_specific_layers
    modules:
      - model.layers.27
      - model.layers.28
      - model.layers.29
      - model.layers.30
      - model.layers.31
      - model.layers.32
      - model.layers.33
      - model.layers.34
      - model.layers.35
      - model.norm.weight
    lr: 3.5e-5
    weight_decay: 0.0060
plugins: []
logging_steps: 1
save_steps: 100
save_total_limit: 1
include_tkps: true
use_wandb: false
special_tokens:
  pad_token: "<|endoftext|>"

5. Run it.
Example:
accelerate launch -m axolotl.cli.train config.yaml

### Config yaml

```yaml
- Setup with all liger options and enable it.
Example:
base_model: Goekdeniz-Guelmez/Josiefied-Qwen3-4B-Instruct-2507-gabliterated-v2
trust_remote_code: true
tokenizer_use_fast: true
tokenizer_type: AutoTokenizer
model_type: AutoModelForCausalLM
output_dir: /workspace/data/outputs/Qwen3-4B
seed: 42
pretraining_dataset:
  - path: json
    data_files:
      - /workspace/data/datasets/josiefied/*.jsonl
    text_column: text
    split: train
    type: pretrain
dataset_num_proc: 16       
dataset_keep_in_memory: true
shuffle_merged_datasets: true
dataloader_num_workers: 4 
dataloader_pin_memory: true
dataloader_prefetch_factor: 2
dataloader_drop_last: true
sequence_len: 32768
sample_packing: false
pad_to_sequence_len: true

micro_batch_size: 3      
gradient_checkpointing: true  
activation_offloading: false  
fp8: false               
bf16: true

gradient_accumulation_steps: 1
max_steps: 100
tf32: true
flash_attention: true
torch_compile: false       
optimizer: adamw_torch_fused
learning_rate: 5.0e-5
weight_decay: 0.01
max_grad_norm: 1.0
lr_scheduler: one_cycle
lr_scheduler_kwargs:
  pct_start: 0.1
  anneal_strategy: cos
  div_factor: 25.0
  final_div_factor: 10.0
  three_phase: false
warmup_ratio: 0.04
lr_groups:
  - name: embeddings
    modules:
      - model.embed_tokens.weight
    lr: 3.0e-5
    weight_decay: 0.0036
  - name: task_specific_layers
    modules:
      - model.layers.27
      - model.layers.28
      - model.layers.29
      - model.layers.30
      - model.layers.31
      - model.layers.32
      - model.layers.33
      - model.layers.34
      - model.layers.35
      - model.norm.weight
    lr: 3.5e-5
    weight_decay: 0.0060
plugins:
  - axolotl.integrations.liger.LigerPlugin
liger_rope: true
liger_rms_norm: true
liger_glu_activation: true
liger_layer_norm: true
liger_fused_linear_cross_entropy: true
liger_dropout: true
logging_steps: 1
save_steps: 100
save_total_limit: 1
include_tkps: true
use_wandb: false
special_tokens:
  pad_token: "<|endoftext|>"  

- Setup with torch compile enable.
Example:
base_model: Goekdeniz-Guelmez/Josiefied-Qwen3-4B-Instruct-2507-gabliterated-v2
trust_remote_code: true
tokenizer_use_fast: true
tokenizer_type: AutoTokenizer
model_type: AutoModelForCausalLM
output_dir: /workspace/data/outputs/Qwen3-4B
seed: 42
pretraining_dataset:
  - path: json
    data_files:
      - /workspace/data/datasets/josiefied/*.jsonl
    text_column: text
    split: train
    type: pretrain
dataset_num_proc: 16 
dataset_keep_in_memory: true
shuffle_merged_datasets: true
dataloader_num_workers: 4 
dataloader_pin_memory: true
dataloader_prefetch_factor: 2
dataloader_drop_last: true
sequence_len: 32768
sample_packing: false
pad_to_sequence_len: true

micro_batch_size: 1       
gradient_checkpointing: true  
activation_offloading: false  
fp8: false                 
bf16: true

gradient_accumulation_steps: 1
max_steps: 100
tf32: true
flash_attention: true
torch_compile: true
torch_compile_backend: inductor
torch_compile_mode: default
optimizer: adamw_torch_fused
learning_rate: 5.0e-5
weight_decay: 0.01
max_grad_norm: 1.0
lr_scheduler: one_cycle
lr_scheduler_kwargs:
  pct_start: 0.1
  anneal_strategy: cos
  div_factor: 25.0
  final_div_factor: 10.0
  three_phase: false
warmup_ratio: 0.04
lr_groups:
  - name: embeddings
    modules:
      - model.embed_tokens.weight
    lr: 3.0e-5
    weight_decay: 0.0036
  - name: task_specific_layers
    modules:
      - model.layers.27
      - model.layers.28
      - model.layers.29
      - model.layers.30
      - model.layers.31
      - model.layers.32
      - model.layers.33
      - model.layers.34
      - model.layers.35
      - model.norm.weight
    lr: 3.5e-5
    weight_decay: 0.0060
plugins: []
logging_steps: 1
save_steps: 100
save_total_limit: 1
include_tkps: true
use_wandb: false
special_tokens:
  pad_token: "<|endoftext|>"
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [x] Windows

### Python Version

Python 3.11.14

### axolotl branch-commit

I don't know, i get it from here. Link: https://hub.docker.com/r/axolotlai/axolotl-cloud/tags

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Sorry we missed this. For any future readers, it seems to be a Liger specific error (or triton), perhaps being incompatible with Blackwell series. In this case, we recommend using CCE for now. If CCE also errors, it's likely triton needing to be updated
